### PR TITLE
Convert ObjectInformation to a builder

### PR DESCRIPTION
**Pull request by MrCreosote**
Pain now = makes it easier to add the admin metadata and any other future new fields

### CODE REVIEW

File: src/us/kbase/workspace/database/ObjectInformation.java [hunk 1]
```diff
@@ -239,54 +188,10 @@ private Reference getLast(final List<Reference> refpath) {
 		return refpath.get(refpath.size() - 1);
 	}
 
-	@Override
-	public String toString() {
```
Comment (MrCreosote) [inline src/us/kbase/workspace/database/ObjectInformation.java:243]: `toString()` is too much of a pain to test given how easy it is to just add it in temporarily when you need it.

File: src/us/kbase/workspace/database/UncheckedUserMetadata.java [hunk 2]
```diff
@@ -18,8 +18,6 @@ public class UncheckedUserMetadata {
 	
 	private Map<String, String> metadata;
 	
-	//TODO CODE empty constructor -> empty metadata
```
Comment (MrCreosote) [inline src/us/kbase/workspace/database/UncheckedUserMetadata.java:21]: YAGNI

**Comment by MrCreosote** [comment]
Not sure why the one test is failing at the set up stage consistently while the other isn't... will look into it tomorrow

**Comment by ialarmedalien** [comment]
The issue with the set-up is due to the version of minio not being available from the supplied link -- the issue is intermittent (maybe a load balancer on the minio website and different dir structures on the different servers behind it?) and I found the quickest and easiest way to fix it was just to add the relevant release of minio to the workspace repo.

go to https://dl.minio.io/server/minio/release/linux-amd64/archive/
search for minio.RELEASE.2019-05-23T00-29-34Z -- nowhere to be found (at least some of the time)
it's now in https://dl.minio.io/server/minio/release/linux-amd64/archive/minio-2019-releases.tar.gz

File: src/us/kbase/workspace/database/ObjectInformation.java [hunk 3]
```diff
@@ -301,71 +206,206 @@ public boolean equals(Object obj) {
...
+					objectName == null ||
+					type == null ||
+					savedDate == null ||
+					version < 1 ||
+					savedBy == null ||
+					ws == null ||
+					chksum == null ||
+					size < 1) {
+				throw new IllegalArgumentException("One or more of the required arguments are "
+						+ "not set. Please check the documentation for the builder.");
```
Comment (ialarmedalien) [inline src/us/kbase/workspace/database/ObjectInformation.java:404]: If you want to be extra accurate in your error messaging...

"One or more of the required arguments are invalid or not set."

(objectId / size / version are set, they just don't pass the tests)

**Review by ialarmedalien — APPROVED** [review]
LGTM.

**Comment by MrCreosote** [comment]
Looks like the tests finally passed

**Pull request merged**